### PR TITLE
Updated docs for DS server capability.

### PR DESCRIPTION
**Pull request by rimashah25**
Updated how-to docs for Delivery Service Required Capability and linked the same (Required Capabilities) in DS form.

Related: https://github.com/apache/trafficcontrol/pull/7224

<hr/>

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Portal

## What is the best way to verify this PR?

Click on links on DS form and ensure it routes to right how-to documentation page and the diagram is also updated.
image

## PR submission checklist
- [ ] This PR has tests 
- [x] This PR has documentation 
- [ ] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: docs/source/admin/quick_howto/server_capabililty.rst [hunk 1]
```diff
@@ -42,18 +42,18 @@ Users with the Operations or Admin :term:`Role` can associate one or more server
 
 .. _delivery_service_required_server_capabilities:
 
-Manage delivery service required server capabilities
-====================================================
-Users with the Operations or Admin :term:`Role` can associate one or more required server capabilities with a :term:`Delivery Service` by navigating to a :term:`Delivery Service` via :menuselection:`Services --> Delivery Services` and using the context menu for the :term:`Delivery Services` table and selecting :menuselection:`Manage Required Server Capabilities` or by navigating to :menuselection:`Services --> Delivery Services --> Delivery Service --> More --> Manage Required Server Capabilities`.
+Configure delivery service required server capabilities
+=======================================================
+Users with the Operations or Admin :term:`Role` can associate one or more required server capabilities with a :term:`Delivery Service` by navigating to a :term:`Delivery Service` via :menuselection:`Services --> Delivery Services` and clicking on Required Capabilities checkbox present on Delivery Services form (HTTPS and DNS)`.
```
Comment (mitchell852) [inline docs/source/admin/quick_howto/server_capabililty.rst:47]: actually this part changed. it should say "Users with the DELIVERY-SERVICE:CREATE or DELIVERY-SERVICE:UPDATE permission can associate..."

**Comment by mitchell852** [comment]
I did not test this but all the changes i requested were made

File: traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.DNS.tpl.html [hunk 2]
```diff
@@ -343,11 +343,14 @@ <h3 ng-if="!open()">Previous Value</h3>
                         </div>
                     </div>
                     <div class="form-group" ng-class="{'has-error': hasError(generalConfig.requiredCapabilities), 'has-feedback': hasError(generalConfig.requiredCapabilities)}">
-                        <label class="has-tooltip control-label col-md-2 col-sm-2 col-xs-12" for="capability">Required Capability
+                        <div class="has-tooltip control-label col-md-2 col-sm-2 col-xs-12" for="capability">
```
Comment (ocket8888) [inline traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.DNS.tpl.html:346]: This `for` attribute needs to be on the `<label>` tag below it

File: traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.HTTP.tpl.html [hunk 3]
```diff
@@ -343,11 +343,14 @@ <h3 ng-if="!open()">Previous Value</h3>
                         </div>
                     </div>
                     <div class="form-group" ng-class="{'has-error': hasError(generalConfig.requiredCapabilities), 'has-feedback': hasError(generalConfig.requiredCapabilities)}">
-                        <label class="has-tooltip control-label col-md-2 col-sm-2 col-xs-12" for="capability">Required Capability
+                        <div class="has-tooltip control-label col-md-2 col-sm-2 col-xs-12" for="capability">
+                            <label>Required Capability</label>
```
Comment (ocket8888) [inline traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.HTTP.tpl.html:347]: Same as above

**Review by ocket8888 — CHANGES_REQUESTED** [review]

**Review by ocket8888 — APPROVED** [review]

**Pull request merged**